test/Parse/ConditionalCompilation/compiler.swift

```diff
@@ -0,0 +1,29 @@
+// RUN: %target-typecheck-verify-swift
```

Comment:
Could this *also* be run in Swift 4 mode, to check that `#if compiler` is not affected?

```diff
 // RUN: %target-typecheck-verify-swift
+// RUN: %target-typecheck-verify-swift -swift-version 4
```